osu.Game/Overlays/Direct/DownloadButton.cs

```diff
@@ -1,76 +1,146 @@
 
         public DownloadButton(BeatmapSetInfo set, bool noVideo = false)
         {
-            BeatmapSetDownloader downloader;
             Children = new Drawable[]
             {
                 downloader = new BeatmapSetDownloader(set, noVideo),
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    CornerRadius = 17,
+                    Masking = true,
+                    Child = background = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                    },
+                },
                 icon = new SpriteIcon
                 {
                     Anchor = Anchor.Centre,
                     Origin = Anchor.Centre,
-                    Size = new Vector2(30),
-                    Icon = FontAwesome.fa_osu_chevron_down_o,
+                    Size = new Vector2(13),
+                    Icon = FontAwesome.fa_download,
                 },
+                checkmark = new SpriteIcon
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    X = 8,
+                    Size = Vector2.Zero,
+                    Icon = FontAwesome.fa_check,
+                }
             };
 
             Action = () =>
             {
-                if (!downloader.Download())
+                if (downloader.DownloadState.Value == BeatmapSetDownloader.DownloadStatus.Downloading)
                 {
                     Content.MoveToX(-5, 50, Easing.OutSine).Then()
                            .MoveToX(5, 100, Easing.InOutSine).Then()
                            .MoveToX(-5, 100, Easing.InOutSine).Then()
                            .MoveToX(0, 50, Easing.InSine);
                 }
-            };
-
-            downloader.Downloaded.ValueChanged += d =>
-            {
-                if (d)
-                    this.FadeOut(200);
+                else if (downloader.DownloadState.Value == BeatmapSetDownloader.DownloadStatus.Downloaded)
+                {
+                    // TODO: Jump to song select with this set when the capability is implemented
+                }
                 else
-                    this.FadeIn(200);
+                {
+                    downloader.Download();
+                }
             };
+
+            downloader.DownloadState.ValueChanged += _ => updateState();
+
+            Colour = Color4.White;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            updateState();
+        }
+
+        [BackgroundDependencyLoader(permitNulls:true)]
+        private void load(OsuColour colours)
+        {
+            this.colours = colours;
         }
 
         protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
         {
-            icon.ScaleTo(0.9f, 1000, Easing.Out);
+            Content.ScaleTo(0.9f, 1000, Easing.Out);
```

Comment:
If you're going to use a scale, you need to set anchor/origin to `Centre` else it will look wrong.